### DISCUSSION TITLE
Easier development workflow

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#775] [General]
![{4326FA21-F2FD-428c-87D2-A91A8AD9D7B2}](https://github.com/goki/gi/assets/19886504/70506ec3-fcf3-41cd-90ad-e270ec3d0fe5)

```go
	repNmaes := []string{
		"gi",
		"gide",
		"etable",
		"grid",
		"pi",
		"spell",
		"gopix",
		"gosl",
		"vgpu",
		"goosi",
		"prof",
		"goki",
		"svg",
		"girl",
		"colors",
		"mat32",
		"gix",
		"xyz",
		"files",
		"taskmanager",
		"fi",
		"glop",
		"go-msi",
		"gti",
		"gsm",
		"vci",
		"laser",
		"grows",
		"greasi",
		"grease",
		"enums",
		"cursors",
		"cam",
		"settings",
		"gear",
		"grail",
		"clock",
		"webki",
		"video",
		"kid",
		"glide",
		"xe",
		"matcolor",
		"ki",
		"grog",
		"icons",
		"ordmap",
		"jsfs",
		"grr",
		"directive",
		"go-tools",
		"gengo",
		"rqlite",
		//"mobile",
		"gorqlite",
		"rqlite-server",
		"kigen",
		"stringer",
		"go-difflib",
		"freetype",
		"gotopy",
	}
```

hey,Is there a way to create a PR on the local machine? What I did before was to clone someone else's repository into my repository, then I submitted it to my repository, and finally I clicked Create PR request in my repository. But I found that you have 60 repositories, forks are too troublesome, and I can't access github every time in China, I have to mirror to gitlab, and then I commit to gitlab first, and then synchronously mirror to github under my repository, and finally create PR. I feel that it is troublesome to participate in your development at the moment, and I also share the pressure of your development

**Comment by ddkwork** [comment]
What's more troublesome is that the repository image function of the free version of gitlab can only be submitted and not pulled, so there are too many steps to create a PR, not to mention, to pull your refresh code has to be cloned directly from GitHub, you have to see two copies of the code, which is inconvenient to operate.

**Comment by kkoreilly** [comment]
I will work on making it easier to make pull requests across multiple repositories soon. In the meantime, if you want to make changes to a specific repository, you can create a pull request on that repository. For the specific issue of gimain usage that it looks like you are facing, I will get rid of gimain usage across all repositories now.

**Comment by ddkwork** [comment]
> I will work on making it easier to make pull requests across multiple repositories soon. In the meantime, if you want to make changes to a specific repository, you can create a pull request on that repository. For the specific issue of gimain usage that it looks like you are facing, I will get rid of gimain usage across all repositories now.

No, that's not what I'm describing, it's easy to just pull the code from your repository: just clone it locally, Goland will open the directory and automatically manage 60 repositories, click pull will pull the refreshed code of 60 repositories at once, and then add it to the workspace and you're done. The confusion I described is that it is troublesome to participate in your development, sorry for my bad English, because multiple people developing multiple repositories must be pulling and pushing and pr at the same time, if you can only pr commit code to your repository after forking, it is cumbersome to commit and pull the refresh code of your repository after forking 60 repositories. On the contrary, just pull your repository to refresh the code, and the local test is as mentioned before, and clicking the button on the goland will refresh all the repository code at once. So the question is: is there an easy way to get involved in your development, such as making multiple repository PRs, and then you check my commit merge operation.

**Reply by kkoreilly** [reply]
I think the difficulty that you are describing is that it is hard to fork 60 repositories and create individual PRs on each one whenever you make changes. If so, I am planning to make a command that makes this process easier soon. If not, please tell me.

**Reply by ddkwork** [reply]
> I think the difficulty that you are describing is that it is hard to fork 60 repositories and create individual PRs on each one whenever you make changes. If so, I am planning to make a command that makes this process easier soon. If not, please tell me.

Yes, it's my confusion that has been faced by more people who have wanted to participate in your development work, thank you.

**Comment by ddkwork** [comment]
> I think the difficulty that you are describing is that it is hard to fork 60 repositories and create individual PRs on each one whenever you make changes. If so, I am planning to make a command that makes this process easier soon. If not, please tell me.

hi , i maked a big rep test:

https://github.com/ddkwork/ux

**Reply by kkoreilly** [reply]
I will consider this.

**Reply by kkoreilly** [reply]
See #782.

**Reply by ddkwork** [reply]
> See #782.

I forked most of the repositories to my github repository yesterday, and then mirrored the forked repositories to gitlab, and I just saw your changes: delete all modules, copy the code for each module to the goki repository and replace the modules with packages, and keep the commit history, well done. Is it possible to archive most of the rest of the warehouses at the moment? In order for more developers to take a look at the repositories that only need to be paid attention to, then I should delete all forks of github and gitlab, and then re-fork goki, files, task manager and other 4 repositories, 4-5 repositories are really much easier than the previous 60, thank you for the changes, I will join your development work as soon as possible.

**Reply by kkoreilly** [reply]
We are still in the process of redesigning the repository structure, and once we are done I will delete the old repositories. There should only be one or two repositories once we are done.
